rust/crd/src/s3logdir.rs

```diff
@@ -0,0 +1,170 @@
+                // !!!!!
+                return Ok(None);
+            };
+
+        // Check that a bucket name has been defined
+        s3bucket
+            .as_ref()
+            .and_then(|i| i.bucket_name.as_ref())
+            .context(BucketNameMissingSnafu)?;
+
+        if let Some(conn) = s3bucket.as_ref().and_then(|i| i.connection.as_ref()) {
+            if let Some(tls) = &conn.tls {
+                match &tls.verification {
+                    TlsVerification::None {} => return S3TlsNoVerificationNotSupportedSnafu.fail(),
+                    TlsVerification::Server(server_verification) => {
+                        match &server_verification.ca_cert {
+                            CaCert::WebPki {} => {}
+                            CaCert::SecretClass(_) => {
+                                return S3TlsCaVerificationNotSupportedSnafu.fail()
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if let Some(conn) = s3bucket.as_ref().and_then(|i| i.connection.as_ref()) {
+            if conn.tls.as_ref().is_some() {
+                tracing::warn!("The resource indicates S3-access should use TLS: TLS-verification has not yet been implemented \
+            but an HTTPS-endpoint will be used!");
+            }
+        }
+        Ok(Some(S3LogDir {
+            bucket: s3bucket.unwrap(),
+            prefix,
+        }))
+    }
+
+    /// Constructs the properties needed for loading event logs from S3.
+    /// These properties are later written in the `HISTORY_CONFIG_FILE_NAME_FULL` file.
+    ///
+    /// The following properties related to credentials are not included:
+    /// * spark.hadoop.fs.s3a.aws.credentials.provider
+    /// * spark.hadoop.fs.s3a.access.key
+    /// * spark.hadoop.fs.s3a.secret.key
+    /// instead, the environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are set
+    /// on the container start command.
+    pub fn history_server_spark_config(&self) -> BTreeMap<String, String> {
+        let mut result = BTreeMap::new();
+
+        result.insert("spark.history.fs.logDirectory".to_string(), self.url());
+
+        if let Some(endpoint) = self.bucket.endpoint() {
+            result.insert("spark.hadoop.fs.s3a.endpoint".to_string(), endpoint);
+        }
+
+        if let Some(conn) = self.bucket.connection.as_ref() {
+            if let Some(S3AccessStyle::Path) = conn.access_style {
+                result.insert(
+                    "spark.hadoop.fs.s3a.path.style.access".to_string(),
+                    "true".to_string(),
+                );
+            }
+        }
+        result
+    }
+
+    pub fn application_spark_config(&self) -> BTreeMap<String, String> {
+        let mut result = BTreeMap::new();
+        result.insert("spark.eventLog.enabled".to_string(), "true".to_string());
+        result.insert("spark.eventLog.dir".to_string(), self.url());
+        result
```

Comment:
I think we need to configure endpoint, credentials, access style etc. here
Recall the ADR

> The operator will read the s3Connection and set up the fs.s3a.aws.credentials.provider and co (endpoint, accesskey, secretkey, tls, path-style access - basically all attributes of S3Connection) settings. Afterwards if logFileDirectory attribute is set: Set fs.s3a.bucket.<logFileDirectory-bucket-name(here spark-eventlogs)>.aws.credentials.provider and co to overwrite endpoint and credentials for the logging bucket. Set spark.eventLog.enabled property to true and will construct the spark.eventLog.dir from s3a://<logFileDirectory-bucket-name(here spark-eventlogs)>/<logFileDirectory-prefix(here eventlogs/)>.

Reply:
Done.